### PR TITLE
Psp 3890

### PR DESCRIPTION
**Pull request by devinleighsmith**
Update research properties on map after modification.
Fix warnings on generic modal.
remove code from deprecated useApi hook.
prevent map properties from being added with duplicate identifiers
remove dead code in model
prevent duplicate fetch research file requests
add context/hook to manage map properites.
remove dead map prop code.

### CODE REVIEW

File: frontend/src/components/common/GenericModal.tsx [hunk 1]
```diff
@@ -125,7 +126,7 @@ export const GenericModal = (props: BsModalProps & ModalProps) => {
   };
 
   return (
-    <ModalContainer {...props} show={showState}>
+    <ModalContainer {...rest} show={showState}>
```
Comment (devinleighsmith) [inline frontend/src/components/common/GenericModal.tsx:129]: fixes react warnings about unsupported props

File: frontend/src/components/maps/leaflet/InventoryLayer.tsx [hunk 2]
```diff
@@ -132,12 +128,9 @@ export const InventoryLayer: React.FC<InventoryLayerProps> = ({
   zoom,
   filter,
   onMarkerClick,
-  onRequestData,
 }) => {
   const mapInstance = useMap();
-  const [features, setFeatures] = useState<PointFeature[]>([]);
-  const [loadingTiles, setLoadingTiles] = useState(false);
-  const { loadProperties } = useApi();
+  const { search, properties, loading } = useMapSearch();
```
Comment (devinleighsmith) [inline frontend/src/components/maps/leaflet/InventoryLayer.tsx:133]: use the context (via the hook) for the response, loading state.

File: frontend/src/components/maps/leaflet/Map.tsx [hunk 3]
```diff
@@ -91,7 +91,6 @@ const Map: React.FC<MapProps> = ({
   showSideBar,
   whenReady,
   whenCreated,
-  disableMapFilterBar,
```
Comment (devinleighsmith) [inline frontend/src/components/maps/leaflet/Map.tsx:94]: This was unused.

File: frontend/src/components/maps/leaflet/Map.tsx [hunk 4]
```diff
@@ -212,8 +211,8 @@ const Map: React.FC<MapProps> = ({
 
   return (
     <Styled.MapGrid ref={resizeRef} className={classNames('px-0', 'map', { sidebar: showSideBar })}>
-      <LoadingBackdrop show={showLoadingBackdrop} />
-      {!showSideBar && !disableMapFilterBar ? (
+      <LoadingBackdrop show={propertiesLoading} />
```
Comment (devinleighsmith) [inline frontend/src/components/maps/leaflet/Map.tsx:214]: loading state of properties available via the context.

File: frontend/src/features/properties/map/hooks/useMapProperties.tsx [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+import { useApiRequestWrapper } from 'hooks/pims-api/useApiRequestWrapper';
+import { useCallback, useContext } from 'react';
+import { TenantContext } from 'tenants';
+
+export const useMapProperties = () => {
+  const {
+    tenant: { propertiesUrl },
+  } = useContext(TenantContext);
+
+  const loadPropertiesRequest = useCallback(
```
Comment (devinleighsmith) [inline frontend/src/features/properties/map/hooks/useMapProperties.tsx:14]: move this logic out of the deprecated useApi hook.

File: frontend/src/features/properties/map/research/ResearchContainer.tsx [hunk 6]
```diff
@@ -39,18 +39,19 @@ export const ResearchContainer: React.FunctionComponent<IResearchContainerProps>
...
+  const { search } = useMapSearch();
 
   const menuItems = researchFile?.researchProperties?.map(x => getPropertyName(x)) || [];
   menuItems.unshift('RFile Summary');
 
-  useEffect(() => {
-    async function fetchResearchFile() {
-      var retrieved = await retrieveResearchFile(props.researchFileId);
-      setResearchFile(retrieved);
-      setIsDirty(false);
```
Comment (devinleighsmith) [inline frontend/src/features/properties/map/research/ResearchContainer.tsx:50]: avoid using flags to trigger functions where possible. In this case the isDirty flag was causing the research file to be retrieved an additional time during an update.

File: frontend/src/features/properties/map/research/add/models.ts [hunk 7]
```diff
@@ -100,15 +100,4 @@ export class PropertyForm {
       district: toTypeCode(this.districtId),
     };
   }
-
-  public toMapModel(): IMapProperty {
```
Comment (devinleighsmith) [inline frontend/src/features/properties/map/research/add/models.ts:104]: removed dead code.

File: frontend/src/features/properties/selector/MapSelectorContainer.tsx [hunk 8]
```diff
@@ -42,7 +43,17 @@ export const MapSelectorContainer: React.FunctionComponent<IMapSelectorContainer
       <PropertySelectorTabsView
         activeTab={activeSelectorTab}
         setActiveTab={setActiveSelectorTab}
-        MapSelectorView={<PropertyMapSelectorFormView onSelectedProperty={onSelectedProperty} />}
+        MapSelectorView={
+          <PropertyMapSelectorFormView
+            onSelectedProperty={(property: IMapProperty) =>
```
Comment (devinleighsmith) [inline frontend/src/features/properties/selector/MapSelectorContainer.tsx:48]: neither the search based property selector or the map based property selector should allow a duplicate to be selected.

File: frontend/src/hooks/useApi.ts [hunk 9]
```diff
@@ -84,44 +67,6 @@ export const useApi = (): IPimsAPI => {
     return axios;
   }, [dispatch]);
 
-  const isPidAvailable = useCallback(
```
Comment (devinleighsmith) [inline frontend/src/hooks/useApi.ts:87]: the following two functions are dead code from citz pims.

Comment (FuriousLlama) [reply frontend/src/hooks/useApi.ts:87]: +1 on cleanup!

**Pull request merged**